Mention configuration-cache types moving to internal package in the upgrade guide (#29447)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

We believe this situation is very unlikely to occur in real projects, as IDE integration and test execution would not work correctly.
If you need to use the Gradle API, declare a `gradleApi` dependency or use the `java-gradle-plugin` plugin.

> ==== Configuration cache implementation packages now under `org.gradle.internal`

Author: @bamboo I removed any references to specific packages and am stating all packages are gone

> References to Gradle types that are not API should be avoided at all costs, as their direct use is definitely unsupported,

Reviewer: ```suggestion
References to Gradle types that are not part of the public API should be avoided, as their direct use is unsupported,
```

> and Gradle internal implementation classes may suffer breaking changes (or be renamed or removed) from a version to another without any warning.

Reviewer: ```suggestion
and Gradle internal implementation classes may suffer breaking changes (or be renamed or removed) from one version to another without warning.
```

> It is important that users are able to tell which parts of the Gradle codebase are API, and which parts are internal.

Reviewer: ```suggestion
It is important for users to distinguish between the API and internal parts of the Gradle codebase.
```

> One way that is achieved is by having `internal` in implementation package names.

Reviewer: ```suggestion
This is typically achieved by including `internal` in the implementation package names.
```

> However, the configuration cache subsystem was not following that pattern before this release, which could be confusing to users.

Reviewer: ```suggestion
However, before this release, the configuration cache subsystem did not follow this pattern, potentially causing confusion.
```

> In order to address that, in the present release, all code originally under `org.gradle.configurationcache*` packages

Reviewer: ```suggestion
To address this issue, in the current release, all code originally under the `org.gradle.configurationcache*` packages
```

> (which was never API) has been moved to new internal (`org.gradle.internal.*`) packages.

Reviewer: ```suggestion
(which was never API) has been moved to new internal packages (`org.gradle.internal.*`).
```
